generate_assets_metadata.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# Generates the metadata of the assets (ai and wasm). If an asset of the `ai_assets` array
+# contains a `chunk_size` key, the script splits the asset into chunks where each chunk has
+# a maximum size of `chunk_size`. The format of the `chunk_size` value is equivalent to the
+# `SIZE` argument in `split` or `gsplit` on macOS. See `split`/`gsplit` man page for more details.
+#
+# The script needs to be executed in the root of the repository.
+#
+# Usage:
+# ./generate_assets_metadata <path of assets_manifest.json> [<wasm version>] [<path of the wasm output directory>]
+
+OUT_DIR="out"
+ASSETS_METADATA_PATH=$OUT_DIR/assets_metadata.json
+
+if [ -z ${GITHUB_ACTIONS} ]; then
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        if command gsplit --version /dev/null; then
+            SPLIT="gsplit"
+        else
+            echo "Requires the GNU version of 'split'. Use 'brew install coreutils' to install it."
+            exit 1
+        fi
+    else
+        SPLIT="split"
+    fi
+else
+    if [ $RUNNER_OS == "macOS" ]; then
+        brew install coreutils
```

Comment:
Do we want to do it here instead of the CI and maybe caching it? This will also allow us to delete this branch an leave only the previous check.

Reply:
Do we need to run this on the mac ci? This is only needed when we push to the release repo and can be done on the ubuntu instance.

Reply:
Unfortunately yes, we need it when we build the iOS flutter example

Reply:
I moved the installation of `coreutils` into the action